.github/workflows/benchmarks.yml

```diff
@@ -150,15 +183,50 @@ jobs:
 
       - name: Upload benchmark result
         uses: actions/upload-artifact@v4
-        if: always()
         with:
           name: benchmark-result
           path: ${{ env.WORKING_DIRECTORY }}/${{ env.BENCHMARK_RESULT }}
           if-no-files-found: error
 
+      # The next section injects CPU profile collected by apmbench into the build.
+      # By copying the profile, uploading it to the artifacts and pushing it
+      # via a PR to update default.pgo.
+
+      - name: Copy CPU profile
+        run: make cp-cpuprof
+      
+      - name: Upload CPU profile
+        uses: actions/upload-artifact@v4
+        with:
+          name: cpu-profile
+          path: ${{ env.WORKING_DIRECTORY }}/${{ env.BENCHMARK_CPU_OUT }}
+          if-no-files-found: error
+
+      - name: Open PGO PR
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          if [ "$RUN_STANDALONE" = true ]; then
+            cd "${{ github.workspace }}"
+            git config user.email "apm@elastic.co"
+            git config user.name "APM Server"
+            git fetch origin main
+            git checkout main
+            BRANCH="update-pgo-$(date +%s)"
+            git checkout -b "$BRANCH"
+            mv "$PROFILE_PATH" x-pack/apm-server/default.pgo
+            git add x-pack/apm-server/default.pgo
+            git commit -m "PGO: Update default.pgo from benchmarks $WORKFLOW."
+            git push -u origin "$BRANCH"
+            gh pr create -B main -H "$BRANCH" -t "PGO: Update default.pgo" -b "Update default.pgo CPU profile from the benchmarks [workflow]($WORKFLOW)." -R elastic/apm-server
+          fi
+        env:
+          PROFILE_PATH: ${{ env.WORKING_DIRECTORY }}/${{ env.BENCHMARK_CPU_OUT }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This will not trigger a build when the PR is created. Unfortunately, GitHub does not trigger builds if the actor is a `github-actions bot`. For such, there are two alternatives:

1) Use PAT or finer-grained tokens
2) Or a GitHub app to create ephemeral tokens.


`1)` is not anymore what we use, but `2)`, you can use the GitHub app as done in https://github.com/elastic/apm-server/pull/14217/files#diff-68bec17d560c218f68b50a1e84ac73026a82e64927e706cffff69501cd151970R60-R83

Reply:
@v1v thank you for pointing to this, I updated the workflow and it uses the same plumbing to setup all necessary pieces to prepare the github config and the access token. Now the runs work as expected see [link](https://github.com/elastic/apm-server/actions/runs/11134770091/job/30943589134) and the raised [PR](https://github.com/elastic/apm-server/pull/14229) has properly signed commits.